gettext-sys/build.rs

```diff
@@ -191,7 +191,7 @@ fn main() {
     let mut cmd = Command::new("tar");
     cmd.current_dir(&build_dir.join("gettext"))
         .arg("xJf")
-        .arg(&src.join("gettext-0.21.tar.xz"))
+        .arg(&src.join("gettext-0.22.5.tar.xz"))
```

Comment:
I guess this update will allow removing the workaround above:
```
    // Work around build failure with Clang 16, which turned this warning into an error.
    cflags.push("-Wno-incompatible-function-pointer-types");
```

However, not sure if this PR will be merged given that the plan seems to be to move away from the included source tar archives anyway.
@Minoru If these changes are wanted, let me know and I can update this PR.

Reply:
> I'm keeping this open for two reasons:
> 
> 1. I'm having trouble with moving away from the tarballs, so I might just merge this if I get stuck. Will post about it once I've tried all the ideas I have
> 2. even if I do remove the tarball, I still want to nick the fixes for the tests that you've made here :)

Why not merge followed by attempting to do what you want(I don't really see the difference)
Some other rust projects that depend on this crate would be immediately fixed to compile on clang 18 once this is merged

Currently, I am packaging in termux packages repo to use a custom fork of gettext-rs rather than this original tepo solely because the old gettext source code is broken and can't compile properly on newer clang versions.

Reply:
Yeah, I've been dragging my feet because I thought I'm almost done with the tarball stuff, but it's been going for a few weeks now, so you're probably right. I'll see to it next week, or maybe even this weekend.

Reply:
I removed the workaround in c24e78d657aab13d7f08b37075a4de805f799c4e, let's see what CI thinks of that.